staging/src/k8s.io/api/admissionregistration/v1alpha1/types.go

```diff
@@ -130,21 +133,23 @@ type ValidatingAdmissionPolicyList struct {
 
 // ValidatingAdmissionPolicySpec is the specification of the desired behavior of the AdmissionPolicy.
 type ValidatingAdmissionPolicySpec struct {
-	// ParamSource specifies the kind of resources used to parameterize this policy which has to be a CRD.
+	// ParamKind specifies the kind of resources used to parameterize this policy.
 	// If absent, there are no parameters for this policy and the param CEL variable will not be provided to validation expressions.
-	// If ParamSource refers to a non-existent kind, this policy definition is mis-configured and the FailurePolicy is applied.
+	// If ParamKind refers to a non-existent kind, this policy definition is mis-configured and the FailurePolicy is applied.
 	// +optional
-	ParamSource *ParamSource `json:"paramSource,omitempty" protobuf:"bytes,1,rep,name=paramSource"`
+	ParamKind *ParamKind `json:"paramKind,omitempty" protobuf:"bytes,1,rep,name=paramKind"`
 
 	// MatchConstraints specifies what resources this policy is designed to validate.
-	// The AdmissionPolicy cares about a validation if it matches _any_ Constriant.
+	// The AdmissionPolicy cares about a request if it matches _any_ Constraint.
 	// However, in order to prevent clusters from being put into an unstable state that cannot be recovered from via the API
-	// ValidatingAdmissionPolicy cannot match ValidatingWebhookConfiguration, MutatingWebhookConfiguration, ValidatingAdmissionPolicy, ValidatingAdmissionPolicyBinding, or policy param resources.
+	// ValidatingAdmissionPolicy cannot match ValidatingAdmissionPolicy and ValidatingAdmissionPolicyBinding.
 	// Required.
 	MatchConstraints *MatchResources `json:"matchConstraints,omitempty" protobuf:"bytes,2,rep,name=matchConstraints"`
```

Comment:
@`//+structType=atomic` on the entire matchConstraints? (note that this applies recursively, so it would be unnecessary to put any listType/mapType/structType tags on the contents of MatchResources if we do this)

cc @liggitt for thoughts. The alternative would be to stay granular at this level and make the fields of MatchResources atomic. But I favor making the entire MatchConstraints atomic. It seems risky to have uncoordinated changes across any of the constraints. Note that bindings can be modified separate from policy, even if we do this.

Reply:
Would it be possible to have some of the field defined separately? like user might want a specific resource be excluded from all policy and add this constraint separately?

Reply:
An alternative to putting `structType=atomic` on `matchConstraints` would be to put it on all the fields of `matchConstraints`. This would allow, for example, for the ownership of the exclude fields to be separate from the include fields..

Reply:
since policy binding is what we're expecting multiple actors to use to collaborate or subscope a policy, I think matchConstraints being atomic makes sense